src/Symfony/Component/Console/Tests/Question/ConfirmationQuestionTest.php

```diff
@@ -0,0 +1,62 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Console\Tests\Question;
+
+use PHPUnit\Framework\TestCase;
+use Symfony\Component\Console\Question\ConfirmationQuestion;
+
+class ConfirmationQuestionTest extends TestCase
+{
+    /**
+     * @dataProvider normalizerUsecases
+     */
+    public function testDefaultRegexUsecases($default, $answers, $expected, $message)
+    {
+        $sut = new ConfirmationQuestion('A question', $default);
+
+        foreach ($answers as $answer) {
+            $normalizer = $sut->getNormalizer();
+            $actual = $normalizer($answer);
+            $this->assertEquals($expected, $actual, sprintf($message, $answer));
+        }
+    }
+
+    public function normalizerUsecases()
+    {
+        return [
+            [
+                true,
+                ['y', 'Y', 'yes', 'YES', 'yEs', ''],
+                true,
+                'When default is true, the normalizer must return true for "%s"',
+            ],
+            [
+                true,
+                ['n', 'N', 'no', 'NO', 'nO', 'foo', '1', '0'],
```

Comment:
IMO `1` should keep being considered as `true` in all cases, and making it `false` now feels risky for existing apps. 
I suggest to change ConfirmationQuestion:51 so that it takes it into account e.g.
`$answerIsTrue =  '1' == $answer || preg_match($regex, $answer);`

Reply:
I can see where you're coming from, but `1` never was considered true when the default is true - anything that wasn't an empty string or 'y' would be considered a falsely answer (I am only testing the default regex here) - so changing this now would be altering behaviour and increasing risk.
The only change made to this test to support my change in `ConfirmationQuestion` was the addition of '0' on the line above, so I think it would be safer to leave it as it is.

Reply:
Indeed, you're right.